src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using SigSpec.Core;
+using Xunit;
+
+namespace SigSpec.CodeGeneration.CSharp.Tests
+{
+	public class GeneratorTests
+	{
+		private readonly SigSpecGeneratorSettings _settings = new SigSpecGeneratorSettings();
+		private readonly SigSpecGenerator _generator;
+
+		private readonly SigSpecToCSharpGeneratorSettings _codeGeneratorSettings = new SigSpecToCSharpGeneratorSettings
+		{
+			CSharpGeneratorSettings = { GenerateNullableReferenceTypes = false }
+		};
+
+		private readonly SigSpecToCSharpGenerator _codeGenerator;
+
+		public GeneratorTests()
+		{
+			_generator = new SigSpecGenerator(_settings);
+			_codeGenerator = new SigSpecToCSharpGenerator(_codeGeneratorSettings);
+		}
+
+		[Theory]
+		[InlineData(typeof(HubWithString), "string")]
+		[InlineData(typeof(HubWithStringOptional), "string")]
+		[InlineData(typeof(HubWithInt), "int")]
+		[InlineData(typeof(HubWithIntOptional), "int")]
```

Comment:
Value types should still get a ? in this test I think (or be `Nullable<T>`).
(int, Guid, Struct)
They are value types, so `GenerateNullableReferenceTypes = false` shouldn't stop them.

Reply:
I'm not sure about having that inconsistency then? 
Perhaps re-add our own options flag and use that for generating nullable or not.

Reply:
As discussed IRL (for anyone else following along).
Nullable value types are an old c# feature, nullable reference types are a new c# feature.
The `GenerateNullableReferenceTypes = false` option is for older c# back compatibility.